janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/step/JanusGraphStep.java

```diff
@@ -202,22 +216,30 @@ public String toString() {
         if (hasLocalContainers.isEmpty()) {
             return StringFactory.stepString(this, Arrays.toString(this.ids), hasContainers);
         }
-        if (hasLocalContainers.size() == 1){
+        if (hasLocalContainers.size() == 1 && hasLocalContainers.values().iterator().next().size() == 1) {
+            Map<List<HasContainer>, QueryInfo> localContainers = hasLocalContainers.values().iterator().next();
```

Comment:
I think it's better to use a single iterator here. I.e. something like:
```
Map<List<HasContainer>, QueryInfo> singleLocalContainers = hasLocalContainers.values().iterator().next();
if (hasLocalContainers.size() == 1 && singleLocalContainers.size() == 1) {
    final List<HasContainer> containers = new ArrayList<>(hasContainers);
    containers.addAll(singleLocalContainers.keySet().iterator().next());
    return StringFactory.stepString(this, Arrays.toString(this.ids), containers);
}
```

Reply:
good point, done